main/src/library/Vector.flix

```diff
@@ -285,15 +287,23 @@ mod Vector {
     /// Returns `None` if `v` is empty.
     ///
     pub def head(v: Vector[a]): Option[a] =
```

Comment:
Performance for Vector is important, so here I would do the bounds check directly (i.e. that length != 0) and then return None/Some based on that.